src/main/java/bio/terra/service/search/SearchService.java

```diff
@@ -25,4 +43,35 @@ public SearchIndexModel indexSnapshot(Snapshot snapshot, SearchIndexRequest sear
         List<Map<String, Object>> values = bigQueryPdao.getSnapshotTableData(snapshot, searchIndexRequest.getSql());
         return new SearchIndexModel();
     }
+    public SearchQueryResultModel querySnapshot(
+            SearchQueryRequest searchQueryRequest, List<String> indicesToQuery,
+            @Valid Integer offset, @Valid Integer limit
```

Comment:
I'm curious was benefit `@Valid` provides here. Also, is `null` permitted for these arguments?

Reply:
I believe if values are not provided, spring will fill these in with their default values, so they will not be null

Reply:
I agree, I don't think they can be null. So I would change this code to use `int` instead of `Integer` since the caller is ensuring that they are non-null before they're passed to this layer of the API. Leaving them as `Integer` means we'd have to handle `null` here, which we shouldn't need to do.

Regarding `@Valid`, I think it should be removed unless we know why we're adding it here.

Reply:
I dont think I can change this, the method header is set by spring.